src/app/@core/utils/languages.service.ts

```diff
@@ -5,56 +5,98 @@ import { TranslateService } from '@ngx-translate/core';
+import fr from '@angular/common/locales/fr';
+import ar from '@angular/common/locales/ar';
+registerLocaleData(fr)
+registerLocaleData(ar)
+const langs = ['ar', 'fr'];
+const RtlLangs = ['ar'];
+
+
+/*
+langs.forEach(lang => {
+    const path =`@angular/common/locales/${lang}`; 
+    import(path).then((data)=>registerLocaleData(data));
+});
+*/
+
+const defaultLang = "fr";
+
+export class DynamicLocaleId extends String {
+    constructor(protected service: LanguagesService) {
+        super(defaultLang);
+    }
+
+    toString() {
+        return this.service.currentLang;
+    }
+}
+
 @Injectable()
 export class LanguagesService {
     directions = NbLayoutDirection;
-    langs = ['ar', 'fr'];
-    RtlLangs = ['ar'];
+
     savedLanguage: string;
 
     constructor(private translateService: TranslateService, private directionService: NbLayoutDirectionService, private userService: UserData, private authService: NbAuthService) { }
 
     init() {
-        this.translateService.addLangs(this.langs);
-        this.translateService.setDefaultLang('fr');
+        this.translateService.addLangs(langs);
+        this.translateService.setDefaultLang(defaultLang);
 
         const browserLang = this.translateService.getBrowserLang();
-        console.log('yes init language');
+
         this.authService.isAuthenticated()
-            .pipe(
-                tap(authenticated => {
-                    console.log('au', authenticated);
-                    if (authenticated) {
-                        this.authService.getToken().subscribe(token => console.log(token, 'zzzz'));
-                        console.log('""é');
-                        this.userService.getSetting()
-                            .pipe(map(setting => setting.language))
-                            .subscribe((language) => {
-                                this.savedLanguage = language;
-                                this.changeLanguage(this.savedLanguage ? this.savedLanguage : browserLang.match(/ar|fr/) ? browserLang : 'fr');
-                            });
-                    } else {
-                        this.savedLanguage = browserLang.match(/ar|fr/) ? browserLang : 'fr';
-                        this.changeLanguage(this.savedLanguage);
-                    }
-                }),
-            );
+            .subscribe((authenticated) => {
+                if (authenticated) {
+                    this.userService.getSetting()
+                        .pipe(map(setting => setting.language))
+                        .subscribe((language) => {
+                            this.savedLanguage = language;
+                            this.changeLanguage(this.savedLanguage ? this.savedLanguage : browserLang.match(this.getLangs().join('|')) ? browserLang : defaultLang);
+                        });
+                } else {
+                    this.savedLanguage = browserLang.match(this.getLangs().join('|')) ? browserLang : defaultLang;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/HEPIA-Association/association-WebClient/pullRequest?prid=7914646)
```suggestion
                    this.savedLanguage = browserLang.match(this.getLangs().join("|")) ? browserLang : defaultLang;
```